lib/presentation/widget/shared_space/shared_space_viewmodel.dart

```diff
@@ -390,6 +392,33 @@ class SharedSpaceViewModel extends BaseViewModel {
     });
   }
 
+  void openRenameWorkGroupModal(BuildContext context, SharedSpaceNodeNested sharedSpace) {
+    _appNavigation.popBack();
+
+    EditTextModalSheetBuilder()
+      .key(Key('rename_work_group_modal'))
+      .title(AppLocalizations.of(context).rename_node(AppLocalizations.of(context).workgroup.toLowerCase()))
+      .cancelText(AppLocalizations.of(context).cancel)
+      .onConfirmAction(AppLocalizations.of(context).rename,
+        (value) => store.dispatch(_renameWorkGroupAction(context, value, sharedSpace)))
+      .setErrorString(
+        (value) => getErrorString(context, value))
+      .setTextController(TextEditingController.fromValue(
+      TextEditingValue(
+        text: sharedSpace.name,
+        selection: TextSelection(baseOffset: 0, extentOffset: sharedSpace.name.length)),
+    ))
```

Comment:
it is messy too understand

Reply:
Why? I simply add a TextControlller object to listen for changes in the text in the TextEditer and selectAll text it when it starts to display.